python-lib/osisoft_client.py

```diff
@@ -896,6 +918,88 @@ def close(self):
     # }
 
 
+class OSIsoftBatchWriter(object):
+    # Each row of data added (write_row) first goes in a streak buffer
+    # streak buffer is meant to be used to push a flow of values to one AF path / webid
+    # If the new row concerns another webid, the current streak buffer is flushed into the request buffer
+    # When request buffer is full, it is flushed into the batch endpoint
+
+    # Each write_row call is adds an entry to the responses list
+    # The pointer into that list is past as a _dku_counter parameter, extracted from the requests just before sending the batch
+    # Uppon receiving the batch response, each individual reponse is reordered
+    #   and the status code / error messages are stored into the responses list at the right pointer
+
+    # Possible improvement: flush the responses list (and result writing in the recipe.py) at each _flush_requests to keep memory from going up.
```

Comment:
Did you try the current version with a big amount of rows ? What are the potential issues with the current implem, can the job crash?